08_education/digital_access.qmd

```diff
@@ -0,0 +1,939 @@
+---
+title: "Share of households with broadband access in the home"
+author: "Manu Alcalá Kovalski"
+date: now
+format:
+  html:
+    self-contained: true
+    toc: true
+editor_options: 
+  chunk_output_type: source
+execute:
+  warning: false
+  message: false
+editor:
+  markdown:
+    wrap: 80
+    canonical: true
+---
+
+```{r}
+options(scipen = 999)
+
+library(tidyverse)
+library(urbnthemes)
+library(tidycensus)
+library(censusapi)
+library(glue)
+library(janitor)
+
+set_urbn_defaults(style = "print")
+```
+
+This metric shows the ratio of households with the presence of a computer and a
+type of broadband internet subscription in their household (we calculate the
+overall ratio, as well as by racial subgroup - measured according to the race of
+each head of the household).
+
+## Process
+
+1.  Housekeeping
+
+2.  Pull demographics for Census Places and Census Counties
+
+3.  Clean and reshape to move data into the vars we want
+
+4.  Calculate the digital access metric
+
+5.  Create a data quality flag
+
+6.  Cleaning and export
+
+## Housekeeping
+
+```{r}
+#| label: housekeeping
+# Explore where to pull data from
+acs5_vars <- load_variables(2021, "acs5", cache = TRUE)
+acs1_vars <- load_variables(2021, "acs1", cache = TRUE)
+acs1_subject_vars <- load_variables(2021, "acs1/subject", cache = TRUE)
+```
+
+## Pull demographics for Census Places and Census Counties
+
+```{r}
+#| label: census-pull-function
+pull_census_data <- function(vars, geography, survey, years) {
+  census_data <- map_dfr(years, ~ get_acs(
+    geography = {{ geography }},
+    variables = vars,
+    survey = survey,
+    year = .x,
+    moe_level = 95
+  ) %>%
+    mutate(year = .x) %>%
+    pivot_wider(names_from = variable, values_from = c(estimate, moe))) %>%
+    rename_with(
+      ~ str_remove(.x, "estimate_"),
+      starts_with("estimate_")
+    ) |>
+    clean_names()
+
+
+  if (geography == "county") {
+    census_data <- census_data %>%
+      filter(!str_detect(geoid, "^72"))
+  }
+
+
+  return(census_data)
+}
+```
+
+### Communities overall
+
+-   Unit of analysis: Households,
+-   Survey: ACS 1
+-   Census table: B28003
+-   Years: 2016, 2017, 2018, 2019, 2021, 2022
+
+We are using the 1-yr ACS for the communities overall (not the subgroups), from
+2016 to 2022. This will be based on table B28003.
+
+```{r}
+# "Population" variables refer to households/people in households in a
+# given category.
+
+# "Digital" variables (those with a "digital" prefix) refer to households/people in households in a given category that have a computer with a broadband internet subscription.
+
```

Comment:
Not a need, this is just a preference thing for clarity - I like to add the census codes & their names as a comment for reference before introducing in the code. Like: 
B28003 - Presence of a Computer and Type of Internet Subsciption in Household
_001 - Total
_004 With a Broadband Internet Subscription